index_configuration_tool/main.py

```diff
@@ -97,6 +100,21 @@ def validate_pipeline_config(config: dict):
     validate_plugin_config(config, SINK_KEY)
 
 
+def write_output(yaml_data: dict, new_indices: set, output_path: str):
+    pipeline_config = next(iter(yaml_data.values()))
+    # Endpoint is a tuple of (type, config)
+    source_config = get_supported_endpoint(pipeline_config, SOURCE_KEY)[1]
+    source_indices = source_config.get(INDICES_KEY, dict())
+    included_indices = source_indices.get(INCLUDE_KEY, list())
+    for index in new_indices:
+        included_indices.append({INDEX_NAME_KEY: index})
+    source_indices[INCLUDE_KEY] = included_indices
+    source_config[INDICES_KEY] = source_indices
```

Comment:
I'm not 100% on the semantics of the `get` operator in lines 107 and 108, but in other contexts dictionaries are basically always passed by reference in python, so this modifies the original in `yaml_data`. Is that intentional?

Reply:
Okay, yes, in line 114 I see that it definitely is. Out of curiosity, why do you only operate on the first value in the yaml (line 104)?

Reply:
> Out of curiosity, why do you only operate on the first value in the yaml (line 104)?

The top level object in the Data Prepper pipeline YAML is a human-readable pipeline name (as the key) followed by the actual pipeline configuration (as the value). Since this logic only aims to add an `include` configuration to the `source` plugin, line 104 drops down a level to ignore the pipeline name.